File: app/src/main/java/com/sunny/player/control/IVideoControl.java
package com.sunny.player.control;

import android.net.Uri;
import android.view.View;

import com.sunny.player.view.SunMediaPlayer;

import java.util.Map;

public interface IVideoControl {

    void setVideoViewListener(IVideoViewListener listener);

    View getVideoView();

    SunMediaPlayer getMediaPlayer();

    void setVideoPath(Uri uri, Map<String, String> headers);

    void startPlay();

    void pausePlay();

    void seekTo(int position);

    int getVideoWidth();

    int getVideoHeight();

    int getVideoDuration();

    int getCurrentPosition();

    boolean isPlaying();

    boolean inPlaybackState();
}

File: app/src/main/java/com/sunny/player/control/IVideoControl.kt
package com.sunny.player.control

import android.net.Uri
import android.view.View
import com.sunny.player.view.SunMediaPlayer

interface IVideoControl {
    fun setVideoViewListener(listener: IVideoViewListener)

    val videoView: View

    val mediaPlayer: SunMediaPlayer?

    val videoWidth: Int

    val videoHeight: Int

    val videoDuration: Int

    val currentPosition: Int

    val isPlaying: Boolean

    fun setVideoPath(uri: Uri, headers: Map<String, String>?)

    fun startPlay()

    fun pausePlay()

    fun stopPlay()

    fun seekTo(position: Int)


    fun inPlaybackState(): Boolean
}